tests/potential/test_potential.py

```diff
@@ -581,5 +581,211 @@ def test_saveload(self):
 
         temp.close()
 
+class QuadPot(potential.PairPotential):
+    """Quadratic potential function used to test potential.Tabulator"""
+
+    def __init__(self, types, params, default={}):
+        super().__init__(types, params, default)
+
+    def _energy(self, r, m, **params):
+        r,u,s = self._zeros(r)
+        u = m*(3-r)**2
+        if s:
+            u = u.item()
+        return u
+
+    def _force(self, r, m, **params):
+        r,f,s = self._zeros(r)
+        f = 2*m*(3-r)
+        if s:
+            f = f.item()
+        return f
+
+    def _derivative(self, param, r, **params):
+        r,d,s = self._zeros(r)
+        if param == 'm':
+            d = (3-r)**2
+        if s:
+            d = d.item()
+        return d
+
+class test_Tabulator(unittest.TestCase):
+    """Unit tests for potential.Tabulator"""
+
+    def test_init(self):
+        """Test creation of object with data"""
+        #test creation with required parameters
+        t = potential.Tabulator(nbins=5, rmin=0.5, rmax=1.5)
+        self.assertEqual(t._nbins, 5)
+        self.assertAlmostEqual(t._rmin, 0.5)
+        self.assertAlmostEqual(t._rmax, 1.5)
+        self.assertAlmostEqual(t.dr, 0.2)
+        np.testing.assert_allclose(t.r, np.linspace(0.5,1.5,6))
+
+        #test creation with required params, fmax, fcut
+        t = potential.Tabulator(nbins=2, rmin=1.0, rmax=2.0, fmax=1.5, fcut=1.0)
+        self.assertEqual(t._nbins, 2)
+        self.assertAlmostEqual(t._rmin, 1.0)
+        self.assertAlmostEqual(t._rmax, 2.0)
+        self.assertAlmostEqual(t.fmax, 1.5)
+        self.assertAlmostEqual(t.fcut, 1.0)
+        self.assertAlmostEqual(t.dr, 0.5)
+        np.testing.assert_allclose(t.r, np.linspace(1.0,2.0,3))
+
+        #test creation with all params and edges=False
+        t = potential.Tabulator(nbins=4, rmin=1.0, rmax=2.5, fmax=1.75, fcut=2.0, edges=False)
+        self.assertEqual(t._nbins, 4)
+        self.assertAlmostEqual(t._rmin, 1.0)
+        self.assertAlmostEqual(t._rmax, 2.5)
+        self.assertAlmostEqual(t.fmax, 1.75)
+        self.assertAlmostEqual(t.fcut, 2.0)
+        self.assertAlmostEqual(t.dr, 0.375)
+        np.testing.assert_allclose(t.r, 1.0+0.375*(np.arange(4)+0.5))
+
+        #test creation with invalid nbins/rmin/rmax setup
+        with self.assertRaises(ValueError):
+            t = potential.Tabulator(nbins=2.5, rmin=0.5, rmax=1.5)
+        with self.assertRaises(ValueError):
+            t = potential.Tabulator(nbins=2, rmin=1.0, rmax=0.5)
+        with self.assertRaises(ValueError):
+            t = potential.Tabulator(nbins=2, rmin=-1.0, rmax=1.0)
+
+    def test_potential(self):
+        """Test energy and force methods"""
+        p1 = QuadPot(types=('1',), params=('m',), default={'m':2})
+        p2 = QuadPot(types=('1','2'), params=('m',), default={'m':1})
+        p_all = [p1, p2]
+
+        #test energy method
+        #test with edges=True
+        t = potential.Tabulator(nbins=4, rmin=1.0, rmax=5.0)
```

Comment:
With the proposed changes, it should be possible for you to make just one instance of the tabulator (you don't need to test `edges` anymore), and then reuse it for the tests. I'd like to see this tried, since it is a more realistic use case then making a new one each time.